index.js

```diff
@@ -578,7 +578,7 @@ class LatticeKeyring extends EventEmitter {
       // the device is unplugged it will time out and we don't need to wait
       // 2 minutes for that to happen.
       this.sdkSession.timeout = CONNECT_TIMEOUT;
-      await this.sdkSession.connect(this.creds.deviceID)
+      return this.sdkSession.connect(this.creds.deviceID)
```

Comment:
Why are you returning this now? Does that mean `finally` won't get hit?

Reply:
If I return the promise, then I can get the `isPaired` return value without having to set a variable here and then return that. It also lets us handle the `.connect` call in the function that calls `_connect` so we could chain `.then()` or `.catch()` calls after `_connect` now.   
  
`finally` always fires.   
  
For both questions, I threw together a minimal example of the same code paths here: [https://codesandbox.io/s/charming-pateu-eldrqs?file=/src/index.js](https://codesandbox.io/s/charming-pateu-eldrqs?file=/src/index.js) If you look at the CodeSandbox console when this runs, you see that the finally block is always logged. Then if you comment out line 16 and uncomment line 17, you can see how we can pass around the `Promise` to catch errors where we want.